types/forge-viewer/index.d.ts

```diff
@@ -26,2432 +27,2513 @@
+    namespace UI {
+      interface DockingPanelOptions {
+        localizeTitle?: boolean;
+        [key: string]: any;
+      }
 
-        enum ViewableType {
-          SPRITE = 1,
-          GEOMETRY = 2
-        }
+      interface ScrollContainerOptions {
+        left?: boolean;
+        heightAdjustment?: number;
+        marginTop?: number;
+        [key: string]: any;
+      }
 
-        class CustomViewable {
-          constructor(position: THREE.Vector3, style: ViewableStyle, dbId: number);
+      interface ContentSize {
+        height: number;
+        width: number;
+      }
 
-          get dbId(): number;
-          get position(): THREE.Vector3;
-          get style(): ViewableStyle;
-        }
+      interface ResizeOptions {
+        maxHeight: number;
+        [key: string]: any;
+      }
 
-        class LevelRoomsMap {
-          addRoomToLevel(levelName: string, room: Room): void;
-          getRoomsOnLevel(levelName: string, onlyRoomsWithDevices: boolean): Room[];
-        }
+      interface AddPropertyOptions {
+        localizeCategory: boolean;
+        localizeProperty: boolean;
+        [key: string]: any;
+      }
 
-        class ModelStructureInfo {
-          model: Viewing.Model;
-          rooms: Room[];
+      interface ControlOptions {
+        collapsible?: boolean;
+        [key: string]: any;
+      }
 
-          constructor(model: Viewing.Model);
+      interface AddControlOptions {
+        index?: number;
+        [key: string]: any;
+      }
 
-          generateSurfaceShadingData(devices: object[], levels?: LevelRoomsMap): Promise<SurfaceShadingData>;
-          getImmediateChildNodesByName(name: string, parentId?: number): number[];
-          getLevel(room: Room): string;
-          getLevelRoomsMap(keepRoomDetail?: boolean): Promise<LevelRoomsMap>;
-          getRoomList(): Promise<Room[]>;
-        }
+      interface DisplayCategoryOptions {
+        localize?: boolean;
+        [key: string]: any;
+      }
 
-        class Room {
-          constructor(id: number, name: string, bounds: THREE.Box3);
+      interface MenuItem {
+        title: string;
+        target: () => void | MenuItem[];
+      }
 
-          get bounds(): THREE.Box3;
-          get devices(): RoomDevice[];
-          get id(): number;
-          get info(): { properties: any[] };
-          set info(value: { properties: any[] });
-          get name(): string;
+      const COLLAPSED_CHANGED = 'Control.CollapsedChanged';
+      const VISIBILITY_CHANGED = 'Control.VisibilityChanged';
+      const CONTROL_ADDED = 'ControlGroup.ControlAdded';
+      const CONTROL_REMOVED = 'ControlGroup.ControlRemoved';
+      const SIZE_CHANGED = 'ControlGroup.SizeChanged';
+      const STATE_CHANGED = 'Button.StateChanged';
+
+      class DockingPanel {
+        constructor(parentContainer: HTMLElement, id: string, title: string, options?: DockingPanelOptions);
+
+        closer: HTMLElement;
+        container: any;
```

Comment:
Should this be really any?

Reply:
The main issue is that dockRight and dockBottom properties aren't part of HTMLElement, and these are very much needed by Forge viewer apps using panels (well, Dasher, anyway). I'd be happy if there were an easy way to augment the HTMLElement type with these properties. I just defaulted to any, which I agree isn't ideal.